mmv1/products/datafusion/Instance.yaml

```diff
@@ -227,4 +227,28 @@ properties:
           The resource name of the Pub/Sub topic. Format: projects/{projectId}/topics/{topic_id}
         required: true
         input: true            
+  - !ruby/object:Api::Type::Array
+      name: 'accelerators'
+      description: |
+        List of accelerators enabled for this CDF instance.
 
+        If accelerators are enabled it is possible a permadiff will be created with the Options field. 
+        Users will need to either manually update their state file to include these diffed options, or include the field in a [lifecycle ignore changes block](https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#ignore_changes).
+      item_type: !ruby/object:Api::Type::NestedObject
+        properties:
+          - !ruby/object:Api::Type::Enum
+            name: 'acceleratorType'
+            description: |
+              The type of an accelator for a CDF instance.
+            values:
+              - :CDC
+              - :HEALTHCARE
+              - :CCAI_INSIGHTS
+            required: true
+          - !ruby/object:Api::Type::Enum
+            name: 'state'
```

Comment:
drive-by post review comment: We could consider marking this as `required: true` as well, or mark clientside "ENABLED" default. We need more complete guidance on this, but we want to avoid `default_from_api` wherever possible- it creates worse Terraform plans, and can have surprising behaviour when performing complex updates. We still need to use it a lot of the time, though, particularly on already-released fields.

Particularly, there are often cases where making a field (in a new resource) or a subfield (of a new field) required is preferable to mapping to the minimum possible message the API allows. It creates a very clear schema, and makes the user's configuration+plan very explicit about what's happening. That's not an option for new top-level fields in existing resources or subfields in existing fields, since that'd be a breaking change.

Reply:
Is there a general rule for when to approach something with "this supports the main use cases better" as opposed to just matching the API functionality? Or is it just the kind of thing that will make more sense with further experience working with the provider and adding more resources and functionality?

Reply:
Right now just further experience- we need to create guidance/define a general approach, but the rules will always be pretty subjective.

My thinking here is: consider the potential for confusion here if a user has a DISABLED accelerator and the state field is not in config Terraform won't see a diff, and recreating will create a resource in a different state- vs the annoyance of typing (and realistically, copy-pasting) "ENABLED" into configs.